R/utilities.R

```diff
@@ -904,13 +904,16 @@ createMappingList <- function(x, indices)
 }
 
 MIN.NCHAR.FUZZY <- 3L
-#' Creates a logical matrix flagging any pair of strings that have at least one string too short
-#' @param names.to.match The list of strings to check the size is below minimum amount
-#' @param min.nchars The minimum number of characters to allow for a fuzzy match
-#' @noRd
-stringsTooShort <- function(names.to.match) {
+
+#' @importFrom flipTime IsDateTime
+namesAreNotFuzzyMatchable <- function(names.to.match) {
+    # Flag any strings that have at least one string too short
     below.min.nchar <- lapply(names.to.match, function(x) nchar(x) < MIN.NCHAR.FUZZY)
-    outer(below.min.nchar[[1L]], below.min.nchar[[2L]], FUN = "|")
+    # Inspect individual names and check if they form a date, if so flag it.
+    .isDateTime <- function(x) vapply(x, IsDateTime, logical(1L), USE.NAMES = FALSE)
+    is.date.time <- lapply(names.to.match, .isDateTime)
+    either.cond <- mapply(`|`, below.min.nchar, is.date.time, SIMPLIFY = FALSE)
```

Comment:
This is the meat of the PR, it'll inspect each label and check if it is recognised as a date and prevent it from being fuzzy matched as a result.